modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/ClassifyByRangeProcess.java

```diff
@@ -0,0 +1,144 @@
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.process.vector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.factory.CommonFactoryFinder;
+import org.geotools.filter.function.CategorizeFunction;
+import org.geotools.filter.function.RangedClassifier;
+import org.geotools.process.ProcessException;
+import org.geotools.process.factory.DescribeParameter;
+import org.geotools.process.factory.DescribeProcess;
+import org.geotools.process.factory.DescribeResult;
+import org.geotools.process.vector.TransformProcess.Definition;
+import org.geotools.util.Converters;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.AttributeDescriptor;
+import org.opengis.filter.FilterFactory;
+import org.opengis.filter.expression.Expression;
+import org.opengis.filter.expression.Function;
+
+/**
+ * Computes a new attribute to classify another attribute by intervals over vector data sets.
+ * 
+ * @author Mauro Bartolomeoli
+ *
+ * @source $URL$
+ */
+@DescribeProcess(title = "ClassifyByRange", description = "Computes a new attribute to classify another attribute by intervals over vector data sets.")
+public class ClassifyByRangeProcess implements VectorProcess {
+    
+    static FilterFactory filterFactory = CommonFactoryFinder.getFilterFactory(null);
+    
+    @DescribeResult(name = "result", description = "Classified feature collection")
+    public SimpleFeatureCollection execute(
+            @DescribeParameter(name = "features", description = "Input feature collection") SimpleFeatureCollection features,
+            @DescribeParameter(name = "classifyOnAttribute", description = "Attribute to be classified using intervals of values.") String classifyOnAttribute,
+            @DescribeParameter(name = "thresholds", min = 0, description = "Comma delimited list of thresholds (use this one to specify custom intervals).") String thresholds,
+            @DescribeParameter(name = "classifier", min = 0, description = "Classifier type (EqualInterval, Quantile, Jenks). Use with classes to calculate intervals automatically") String classifier,
+            @DescribeParameter(name = "classes", min = 0, description = "Classifier # of classes, use with classifier (defaults to 5).") Integer classes,
+            @DescribeParameter(name = "include", min = 0, defaultValue = "FALSE", description = "Include or exclude current threshold in the interval.") Boolean include,
+            @DescribeParameter(name = "outputAttribute", min = 0, description = "Name of the output attribute with class values (defaults to class).") String outputAttribute,
+            @DescribeParameter(name = "outputValues", min = 0, description = "Comma delimited list of class values for each given threshold (+1 for out of range).") String outputValues,
```

Comment:
Same here. Imho the output values should be close to the thresholds so that they can be read toghether (at least in a Java call, in a WPS call the convention of repeating the input n times will make that hard, but it's how the protocol is designed)